### PR TITLE
Initial commit of nmcli_add_addrs OASIS role

### PR DESCRIPTION
**Pull request by tehsmyers**

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+nmcli_add_addrs_interface: "{{ omit }}"
```
Comment (greg-hellings) [inline defaults/main.yml:6]: I believe, in other places we've run into this type of usage, we've defaulted to nothing here and used the `| default(omit)` filter on the task itself. Still have the variable defined, but just comment it out as such:

`# nmcli_add_addrs_interface: eth0`

@robled and @andrew103 have encountered this before and came up with an existing paradigm. So long as this one matches that, we're all good (and should document it in meta_standards).

File: tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
+- name: Check that either interface or connection name is provided
```
Comment (greg-hellings) [inline tasks/main.yml:1]: I'm just gonna go ahead and assume that all of this is reasonable and sane.

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  - pip --version
+  - pip install ansible ansible-lint yamllint flake8 molecule docker
+  - ansible --version
+  - ansible-lint --version
+  - yamllint --version
+  - flake8 --version
+  - molecule --version
+
+script:
+  - molecule test -s connection_only
```
Comment (greg-hellings) [inline .travis.yml:20]: If we're only using the `connection_only` scenario, do we also need the `docker` one down there?

**Review by greg-hellings — APPROVED** [review]
Conditional approval depending on responses to comments.

File: .travis.yml (hunk 3 above)
Comment (tehsmyers) [reply .travis.yml:20]: Having the docker create/destroy books split out is handy for making it easier to write additional docker scenarios, but it's easy enough to split out later that I'll get rid of it until/unless we actually develop another scenario depending on the docker bits.

File: defaults/main.yml (hunk 1 above)
Comment (tehsmyers) [reply defaults/main.yml:6]: I like that style, and agree that we should document it. In this case, though, it leads to ugly logic in the tasks (`(var | default(omit) != omit`). I've added comments (woops) to `defaults/main.yml` to explain why they default to omit, and also explain in a little more detail how the interface and connection are related to each other, and why at least one must be set.

File: tasks/main.yml (hunk 2 above)
Comment (tehsmyers) [reply tasks/main.yml:1]: I had another scenario that attempted to assert against it, but it proved very difficult to successfully run a role that fails on purpose. I plan to tinker with that a little more, but wanted to get the code up first to make it a little easier to file an issue about it.

File: defaults/main.yml (hunk 1 above)
Comment (greg-hellings) [reply defaults/main.yml:6]: So in the `when` clause and the assert `that` clause, you can directly test "var is defined" as one would do in pseudocode somewhere (analogous to `is not None` in Python or `typeof foo !== 'undefined'` in JavaScript). That allows you to directly put `{{ var | default(omit) }}` into places where it's an optional argument for a parameter and simply leave it undefined in `defaults/main.yml` for places where on sane default exists. You can then test for `is defined` and win at all the things.

**Pull request merged**